src/pickup.cpp

```diff
@@ -169,6 +181,44 @@ interact_results interact_with_vehicle( vehicle *veh, const tripoint &pos,
             veh_tool( "hotplate" );
             return DONE;
 
+        case USE_WASHMACHINE: {
+            bool detergent_is_enough = g->u.crafting_inventory().has_charges( "detergent", 5 );
+            bool filthy_items = false;
+            for( auto &i : veh->get_items( veh->part_with_feature( veh_root_part, "WASHING_MACHINE" ) ) ) {
+                static const std::string filthy( "FILTHY" );
+                if( i.has_flag( filthy ) ) {
+                    i.bday = calendar::turn.get_turn();
```

Comment:
I don't see a contradiction here. There's plenty of ways to check before and set after (two `for` loops is one of them). Even better `std::any_of` to check and a `for` to set.